annotations.md

```diff
@@ -40,6 +40,14 @@ This specification defines the following annotation keys, intended for but not l
     ```
 * **org.opencontainers.image.title** Human-readable title of the image (string)
 * **org.opencontainers.image.description** Human-readable description of the software packaged in the image (string)
+* **org.opencontainers.image.base.digest** [Digest](descriptor.md#digests) of the image this image is based on (string)
+  * This SHOULD be the immediate image sharing zero-indexed layers with the image, such as from a Dockerfile `FROM` statement.
+  * This SHOULD NOT reference any other images used to generate the contents of the image (e.g., multi-stage Dockerfile builds).
+* **org.opencontainers.image.base.ref.name** Image reference of the image this image is based on (string)
+  * This SHOULD be image references in the format defined by [distribution/distribution](https://github.com/distribution/distribution/blob/main/reference/reference.go).
```

Comment:
Di you mean to refer to CNCF distribution or [opencontainers/distribution-spec](https://github.com/opencontainers/distribution-spec)?

I haven't seen other spec references to implementations. 

The lines below seem to give enough context for the format.

Reply:
I wasn't able to find a formal spec describing the format of an image reference (`my.registry/repo/blah/blah:tag` or digest `@sha256:...`). I'm not able to find one in distribution-spec either.

The closest I've been able to find to authoritative description is that code in distribution/distribution.

If it's covered in some spec in OCI, that'd obviously be ideal. If it turns out it isn't, I think that'd be a wortwhile improvement to the specs.

Reply:
> I wasn't able to find a formal spec describing the format of an image reference (my.registry/repo/blah/blah:tag or digest @sha256:...). I'm not able to find one in distribution-spec either.
> The closest I've been able to find to authoritative description is that code in distribution/distribution.
> If it's covered in some spec in OCI, that'd obviously be ideal. If it turns out it isn't, I think that'd be a wortwhile improvement to the specs.

I didn't even know it was defined there, which is why I wrote some blog posts about it a while back.
While I agree we should add a standard definition, I don't know if it's needed here as the example you put below that line suffices, and is better than the more granular description on the other `ref.name`.  I'd still suggest you copy/paste your text to `ref.name` as well.
So, just suggesting you can remove the line that refers to distribution/distribution as we don't have other external references (that I know of), and the annotation definition is already an improvement.

As a separate PR, it would be great to define the element names, but not blocking here.

Reply:
> I'd still suggest you copy/paste your text to ref.name as well.

Do you mean to update the description of the `org.opencontainers.image.ref.name` annotation above? If it's okay, I'd like to maintain the narrow scope of this change to avoid including other already-painted bikesheds. 😄 

> So, just suggesting you can remove the line that refers to distribution/distribution as we don't have other external references (that I know of), and the annotation definition is already an improvement.

As an implementor I think that reference is immensely helpful, and I think it makes sense to have _some_ reference to _some_ semi-authoritative grammar, even if it's not official. My example is just to demonstrate that the full registry name should be included, and isn't intended to fully describe how to construct that string. 

> As a separate PR, it would be great to define the element names, but not blocking here.

I agree, that would be very useful.

Reply:
Hrm I thought that image name ref was defined... Maybe we talked about it so much that it lost track of where to land?

I wish @wking could be summoned to give links to all the discussions around image name ref spec 😹

Reply:
> I'd like to maintain the narrow scope of this change to avoid including other already-painted bikesheds.

Fair enough, once this merges, we can do a clarification PR to change:

> The reference must match the following grammar:
> ref       ::= component ("/" component)*
> component ::= alphanum (separator alphanum)*
> alphanum  ::= [A-Za-z0-9]+
> separator ::= [-._:@+] | "--"

to follow your syntax, with a hopeful naming guideline. 

> As an implementor I think that reference is immensely helpful, and I think it makes sense to have some reference to some semi-authoritative grammar, even if it's not official. 

Conceptually, I agree. I think there's a challenge with a spec referring externally to something that may change. For the same bikeshedding comment above, less=more. Getting in without that reference is a win, and we can add clarity to the example. 

Although, I just noticed it isn't a fully qualified reference, including the tag: `index.docker.io/some/image` should be `docker.io/repo/image:tag`, or to get specific `docker.io/org/baseimage:v1`

Reply:
> Conceptually, I agree. I think there's a challenge with a spec referring externally to something that may change. For the same bikeshedding comment above, less=more. Getting in without that reference is a win, and we can add clarity to the example.

I agree, I think moving the grammar into the spec would be a pure win, and we can remove the external link to reference.go when we do that. In the meantime, having something seems better than making people piece together what a ref name is.

Reply:
> I wasn't able to find a formal spec describing the format of an image reference (`my.registry/repo/blah/blah:tag` or digest `@sha256:...`). I'm not able to find one in distribution-spec either.

Pretty sure it's a gap in the specs that I'd like to see cleaned up too. I've got this old issue asking to better define "reference": https://github.com/opencontainers/distribution-spec/issues/216